app/Actions/Album/PositionData.php

```diff
@@ -22,7 +25,25 @@ public function get(string $albumID, array $data): array
 		$result['id'] = $album->id;
 		$result['title'] = $album->title;
 		$result['photos'] = $photoRelation
-			->with(['album', 'size_variants', 'size_variants.sym_links'])
+			->with([
+				'album' => function (BelongsTo $b) {
+					// The album is required for photos to properly
+					// determine access and visibility rights; but we
+					// don't need to determine the cover and thumbnail for
+					// each album
+					$b->without(['cover', 'thumb']);
+				},
+				'size_variants' => function (HasMany $r) {
+					// The web GUI only uses the small and thumb size
+					// variants to show photos on a map; so we can save
+					// hydrating the larger size variants
+					// this really helps, if you want to show thousands
+					// of photos on a map, as there are up to 7 size
+					// variants per photo
+					$r->whereBetween('type', [SizeVariant::SMALL2X, SizeVariant::THUMB]);
```

Comment:
Sorry for being late on this, but I'm surprised by the choice of sizes here. If `SMALL2X`, then why not also `SMALL`? If `THUMB`, then why `SMALL` sizes at all? And why not `THUMB2X`?

From a quick experiment, it seems that the map overview just uses `THUMB`/`THUMB2X`?

Reply:
Yep, here it is: https://github.com/LycheeOrg/Lychee-front/blob/master/scripts/main/mapview.js#L186-L197

Reply:
Right, I failed to notice that it was a range...

So that overview map will only use the THUMBs I believe, but we still need to fetch all the SMALLs in case the user clicks on any of the thumbs...